scripts/dist-fix.js

```diff
@@ -1,70 +1,26 @@
-const {
-  mkdir,
-  readdir,
-  rename,
-  rm,
-  writeFile,
-  copyFile,
-  readFile,
-  unlink,
-} = require('node:fs/promises')
+// @ts-check
+const {readdir, writeFile, readFile} = require('node:fs/promises')
 const path = require('node:path')
 
 ;(async () => {
   const distPath = path.join(__dirname, '../dist')
-  const distCjsPath = path.join(distPath, 'cjs')
-  const distEsmPath = path.join(distPath, 'esm')
-
-  const [dist, distEsm] = await Promise.all([
-    readdir(distPath),
-    readdir(distEsmPath),
-    rm(distCjsPath, {force: true, recursive: true}),
-  ])
+  const dist = await readdir(distPath)
 
+  // Inject type declarations for deno.
+  // See https://docs.deno.com/runtime/manual/advanced/typescript/types
   await Promise.all([
-    mkdir(distCjsPath),
-    writeFile(
-      path.join(distEsmPath, 'package.json'),
-      JSON.stringify({type: 'module', sideEffects: false}),
-    ),
     ...dist
-      .filter((distFilePath) => distFilePath.match(/\.d\.ts$/))
-      .map((distFilePath) =>
-        copyFile(
-          path.join(distPath, distFilePath),
-          path.join(distEsmPath, distFilePath),
-        ),
-      ),
-    ...distEsm
-      .filter((esmFilePath) => esmFilePath.match(/\.js$/))
-      .map(async (esmFilePath) => {
-        const distEsmFilePath = path.join(distEsmPath, esmFilePath)
-
-        const esmFile = await readFile(distEsmFilePath)
-        const esmFileContents = esmFile.toString()
-
-        const dtsFilePath = `./${esmFilePath.replace('.js', '.d.ts')}`
-
-        const denoFriendlyEsmFileContents = [
-          `/// <reference types="${dtsFilePath}" />`,
-          esmFileContents,
-        ].join('\n')
+      .filter((filePath) => filePath.match(/\.m?js$/))
+      .filter((filePath) => !filePath.startsWith('chunk-'))
```

Comment:
```suggestion
      .filter((filePath) => !filePath.startsWith('chunk-') && filePath.match(/\.m?js$/))
```

Reply:
done

FYI, used the double-filter on purpose. filtering once felt like a micro-improvement, which IMO reads worse once formatted:
```diff
-      .filter((filePath) => filePath.match(/\.m?js$/))
-      .filter((filePath) => !filePath.startsWith('chunk-'))
+      .filter(
+        (filePath) =>
+          filePath.match(/\.m?js$/) && !filePath.startsWith('chunk-'),
+      )
```